src/poll.rs

```diff
@@ -1098,13 +1105,24 @@ impl Poll {
     }
 }
 
+#[cfg(unix)]
+fn registerable(interest: Ready) -> bool {
+    let unixinterest = UnixReady::from(interest);
+    unixinterest.is_readable() || unixinterest.is_writable() || unixinterest.is_aio()
+}
+
+#[cfg(not(unix))]
+fn registerable(interest: Ready) -> bool {
+    interest.is_readable() || interest.is_writable()
+}
+
 fn validate_args(token: Token, interest: Ready) -> io::Result<()> {
     if token == AWAKEN {
         return Err(io::Error::new(io::ErrorKind::Other, "invalid token"));
     }
 
-    if !interest.is_readable() && !interest.is_writable() {
-        return Err(io::Error::new(io::ErrorKind::Other, "interest must include readable or writable"));
+    if !registerable(interest) {
```

Comment:
hmm... ok, so FDs that trigger AIO readiness do not also have readable / writable? I guess this would be fine then...